scripts/hfButler_cpp_clone.py

```diff
@@ -0,0 +1,378 @@
+    """
+    obj.mLinkIds = ids
+    try:
+        obj.mLinkNodes = [obj.podNode.get_link_processor_node(i) for i in ids]
+    except Exception as lExc:
+        click.Abort('Wibulator {} not found in address table'.format(id))
+dpr_mux_choices = {
+    'reset': (0, 0),
+    'playback': (0, 1),
+    'sink': (1, 0),
+    'passthrough': (1, 1)
+}
+# ------------------------------------------------------------------------------
+@link.command('config')
+@click.option('--dr-on/--dr-off', 'dr_on', help='Enable data-reception block', default=None)
+@click.option('--dpr-mux', 'dpr_mux', type=click.Choice(list(dpr_mux_choices.keys()),), help='Configure DPR mux', default=None)
+@click.option('--drop-empty/--keep-empty', 'drop_empty', help='Drop empty hit packets', default=None)
+@click.pass_obj
+def link_config(obj, dr_on, dpr_mux, drop_empty):
+
+    for ln in obj.mLinkNodes:
+        print('>> Link Processor', ln.getId())
+
+        if dpr_mux:
+            dprNode = ln.get_data_router_node().get_dpr_node()
+            print("Configuring DPR mux : ", dpr_mux)
+            dprNode.set_mux_in(dpr_mux_choices[dpr_mux][0], True)
+            dprNode.set_mux_out(dpr_mux_choices[dpr_mux][1], True)
+        
+        if dr_on is not None:
+            print("Configuring data-reception : ", str(dr_on))
+            drNode = ln.get_data_router_node().get_data_reception_node()
+            drNode.getNode("csr.ctrl.en").write(dr_on)
+            drNode.getClient().dispatch()
+
+        if drop_empty is not None:
+            strmArrayNode = ln.get_stream_proc_array_node()
+            strmNode = strmArrayNode.get_stream_proc_node()
+
+            for i in range(obj.mConfigInfo['n_port']):
+                print("Setting drop empty to {} on stream {} ".format(drop_empty, i))
+                strmArrayNode.stream_select(i)
+                strmNode.getNode('csr.ctrl.drop_empty').write(drop_empty)
+            ln.getClient().dispatch()
+
+        print('<< Link Processor ', ln.getId())
+
+# ------------------------------------------------------------------------------
+@link.command('mask')
+@click.option('-p', '--pipes', callback=optionValidators.validate_proc_ids, default='all')
+@click.option('-c', '--chans', callback=optionValidators.validate_chan_ids, default='all')
+@click.argument('action', type=click.Choice(['disable','enable']))
+@click.pass_obj
+def link_mask(obj, pipes, chans, action):
+
+    en_dsbl = True
+    if action == 'disable':
+        en_dsbl = False
+
+    for ln in obj.mLinkNodes:
+        print('>> Link Processor', ln.getId())
+        strmArrayNode = ln.get_stream_proc_array_node()
+        strmNode = strmArrayNode.get_stream_proc_node()
+
+        print('Configuring channel masks in pipelines ', ', '.join((str(p) for p in pipes)))
+        mask_w = 0x0
+        for b in chans:
+            mask_w |= (1 << b)
+
+        for p in pipes:
+            strmArrayNode.stream_select(p)
+            strmNode.set_channel_mask_all(mask_w)
```

Comment:
this sets the same mask on all pipelines, right?  i'm not sure this is desirable.  we should check this replicates old hfButler behaviour, at least.

Reply:
Change made